lib/board.rb

```diff
@@ -0,0 +1,47 @@
+require "pry"
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+  end
+
+  def reset!
+    @cells = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(n)
+    cells[n.to_i - 1]
+  end
+
+  def full?
+    !cells.include?(' ')
+  end
+
+  def turn_count
+    cells.select{|c| c == "X" || c == "O"}.length
+  end
+
+  def taken?(n)
+    index = n.to_i - 1
+    !(cells[index].nil? || cells[index] == " ")
+  end
+
+  def valid_move?(n)
+    !taken?(n.to_i) && n.to_i.between?(1,9)
+  end
+
+  def update(n, player)
+    index = n.to_i - 1
```

Comment:
not entirely necessary to set this as a variable since you are not reusing it in this method.